### ISSUE TITLE
Support for Docker 1.4

### ISSUE THREAD

**Issue by wouterd** [wouterd/docker-maven-plugin#61]
The plugin's unit tests break with Docker 1.4, so it's probable that the plugin doesn't support Docker 1.4.

**Comment by jasperavisser** [comment]
Yeah, I think I can confirm that it doesn't work with 1.4

```
[ERROR] Failed to execute goal net.wouterdanes.docker:docker-maven-plugin:3.0:build-images (build) on project row-change-event-channel: Execution build of goal net.wouterdanes.docker:docker-maven-plugin:3.0:build-images failed: HTTP 500 Internal Server Error -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal net.wouterdanes.docker:docker-maven-plugin:3.0:build-images (build) on project row-change-event-channel: Execution build of goal net.wouterdanes.docker:docker-maven-plugin:3.0:build-images failed: HTTP 500 Internal Server Error
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:224)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
```

Maybe for the meantime you could add a maximum supported version to the README?

**Comment by wouterd** [comment]
Hi Jasper!

Can you tell me what kind of build-images block you have configured? This should still work.
The only real problem seems to be with the way I integration test registry integration. It shouldn't break in production environments.

**Comment by jasperavisser** [comment]
I have this in my pom file:
```
            <plugin>
                <groupId>net.wouterdanes.docker</groupId>
                <artifactId>docker-maven-plugin</artifactId>
                <version>3.0</version>
                <executions>
                    <execution>
                        <id>build</id>
                        <goals>
                            <goal>build-images</goal>
                        </goals>
                        <configuration>
                            <images>
                                <image>
                                    <id>row_change_event_channel</id>
                                    <dockerFile>${project.basedir}/docker/Dockerfile</dockerFile>
                                    <keep>true</keep>
                                    <nameAndTag>haploid/row-change-event-channel:${project.version}</nameAndTag>
                                    <artifacts>
                                        <artifact>
                                            <file>${project.build.directory}/${project.build.finalName}.jar</file>
                                            <dest>/app</dest>
                                        </artifact>
                                        <artifact>
                                            <file>${project.build.directory}/lib</file>
                                            <dest>/app</dest>
                                        </artifact>
                                    </artifacts>
                                </image>
                            </images>
                        </configuration>
                    </execution>
                </executions>
            </plugin>
```

And this in my environment variables (for boot2docker):
```
declare -x DOCKER_CERT_PATH="/Users/haploid/.boot2docker/certs/boot2docker-vm"
declare -x DOCKER_HOST="tcp://192.168.59.103:2376"
declare -x DOCKER_TLS_VERIFY="1"
```

On a ```mvn clean verify``` it fails with:
```
ERROR] Failed to execute goal net.wouterdanes.docker:docker-maven-plugin:3.0:build-images (build) on project row-change-event-channel: Execution build of goal net.wouterdanes.docker:docker-maven-plugin:3.0:build-images failed: HTTP 500 Internal Server Error -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal net.wouterdanes.docker:docker-maven-plugin:3.0:build-images (build) on project row-change-event-channel: Execution build of goal net.wouterdanes.docker:docker-maven-plugin:3.0:build-images failed: HTTP 500 Internal Server Error
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:224)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
...
```

This is on docker 1.4.1.

**Comment by jasperavisser** [comment]
Sorry, I should have also copied:
```
Caused by: javax.ws.rs.InternalServerErrorException: HTTP 500 Internal Server Error
	at org.glassfish.jersey.client.JerseyInvocation.convertToException(JerseyInvocation.java:968)
	at org.glassfish.jersey.client.JerseyInvocation.translate(JerseyInvocation.java:795)
	at org.glassfish.jersey.client.JerseyInvocation.access$500(JerseyInvocation.java:91)
```

I thought for a second that it might've been the / in the ```nameAndTag``` (people forget to escape stuff, sometimes), but it doesn't make a difference if I pick a much simpler name like "bob".

**Comment by wouterd** [comment]
Made a bunch of fixed in 3.1.0 that should fix these server errors. Please let me know if it still breaks.
